src/services/firebase.ts

```diff
@@ -107,9 +115,95 @@ export namespace ourFirebase {
     const matchIds = Object.keys(matchMemberships);
     // TODO: get all matches in one call to firebase, then later call dispatch.
     // Make sure we listen to match changes only once.
-    // '/gamePortal/matches'
-    // dispatch(updateMatchList);
-    getRef('/gamePortal/matches' + matchIds); // TODO
+    // 'gamePortal/matches'
+    // store.dispatch(updateMatchList);
+    let tempMatchIds: Promise<any>[] = [];
+    for (let matchId of matchIds) {
+      // getMatchDetail(matchId).then(() => {tempMatchIds.push(getMatchDetail(matchId))});
+      const match = getMatchDetail(matchId);
+
+      match.then(data => {
+        if (data.status === 'resolved') {
+          // tempMatchIds.push(getMatchDetail(matchId));
+          tempMatchIds.push(match);
+        }
+      });
+    }
+    Promise.all(tempMatchIds)
```

Comment:
You need to push match directly into tempMatchIds. Don't add any match.then as in line 125.

Reply:
```
Promise.all(tempMatchIds).then((data) => {
	// This will filter the promises of any which failed
	failedPromises = datas.filter(data => !(data instanceof Error));
	// Since this is test if failed promises > 0
	// You should fail this test.
	// Continue rest of the code here
});
```

Reply:
Yes, I also considered to do it in this way. However in my opinion, if any of match id fails, the Promise.all will not go to the "then" function, then how to do this filter? Just like in multiple imageUpload function, you asked me to upload the rest of images even some of them failed.

Reply:
Just add a .catch then which will catch all of the failed promises.

Reply:
But you will still need to filter out errors.

Reply:
As far as I known, for tempMatchIds  there will be 2 possible cases:
1. All of them are resolved, then the filter function in the "then" function is meaningless, because there will be no ERROR data.
2. Any of them is rejected, then the catch function will only handle with the first rejected promise. So how can I catch all of the failed promises here?

Reply:
Just handle everything in then. If there are errors you will know, if there are not any errors move further. I am not sure, what is the issue here.